### ISSUE TITLE
move examples from core into correct sub-packages

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1890]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#1890|
|Type|enhancement|
|Created|2006-07-02 15:36:41Z|
|Branch|https://github.com/twisted/twisted/tree/subproject-examples-1890|

e.g. pairudp.py is now in core but ought to be in pair.

Attachments:

* [sub-examples-1890.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/674/6742ebb45229338ec88bed12ed0ef966979b6827/c154e3a5de2fcec336e1573540065d5b97c9c9e2.patch) (1973 bytes) - added by collab on 2008-07-01 23:39:40Z - patch against r24208

<details><summary>Searchable metadata</summary>

```
trac-id__1890 1890
type__enhancement enhancement
reporter__itamarst itamarst
priority__low low
milestone__ 
branch__branches_subproject_examples_1890 branches/subproject-examples-1890
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1151854601000000 1151854601000000
changetime__1219504478000000 1219504478000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

pairudp.py is the most obvious one that should me moved but not sure if there are any others, maybe the cursesclient.py which uses words (although it's a GUI app)? Attached a [[sub-examples-1890.patch](None/sub-examples-1890.patch) patch] for pairudp.py (for some reason svn diff doesn't properly move the file to the new folder, at least thats what the diff shows, hope this works for you).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to collab|
|-|-|

`cursesclient.py` should probably change somehow.  It's supposed to be an example of doing curses with Twisted - maybe it should use something from twisted core to do this, instead of being an IRC client?

It could be a memcache client or an ident client or an ftp client or probably one of a few other things.  It could also completely avoid doing networking and just do some Twistedy stuff with, say, callLater.  Feel like trying to make this change?  We could also keep the current `cursesclient.py` in doc/words/examples, since I guess more examples is generally better than fewer examples.

For the pairdudp.py change, I assume you just did `svn mv doc/core/examples/pairudp.py doc/pair/examples/`?  You're right that `svn diff` doesn't properly reflect moves (and patch doesn't apply the attached diff very well, but I can just do the `svn mv` manually).

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|collab set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_3):
> We could also keep the current `cursesclient.py` in doc/words/examples, since I guess more examples is generally better than fewer examples.
>

I'd vote for moving it to doc/words/examples and giving it a simple description in the docs.

> For the pairdudp.py change, I assume you just did `svn mv doc/core/examples/pairudp.py doc/pair/examples/`?  You're right that `svn diff` doesn't properly reflect moves (and patch doesn't apply the attached diff very well, but I can just do the `svn mv` manually).

Yeah, in that case it's that incorrect svn diff behavior.  I'm probably not able to come up with a better description for that curses client than you can, so assigning it back to you so we can close this ticket after a few simple moves, cause it's been open for more  then years ;)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [24283]) Branching to 'subproject-examples-1890'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner|
|-|-|

Stuff moved, indexes updated.  I'm curious about the comment above, "and giving it a simple description in the docs".  Which docs would benefit from a description?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_7):
> Stuff moved, indexes updated.  I'm curious about the comment above, "and giving it a simple description in the docs".  Which docs would benefit from a description?

The example listings in the .xhtml docs sometimes have a description like [this](http://twistedmatrix.com/projects/core/documentation/examples/index.html) one :

 - echoserv.py - simplest possible TCP server
 - echoserv_udp.py - simplest possible UDP server
 - echoserv_ssl.py - simple SSL server
 - echoclient.py - simple TCP client
 - echoclient_udp.py - simple UDP client

Would be good to have such a one-liner for every example where possible.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun removed owner|
|-|-|

Okay, added

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

I'd say merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [24628]) Merge subproject-examples-1890

Author: exarkun
Reviewer: thijs
Fixes: [#1890](https://github.com/twisted/twisted/issues/1890)

Move two subproject examples out of core into their respective
subprojects.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
